src/uct/base/uct_iface.h

```diff
@@ -526,7 +526,11 @@ typedef struct {
             _priv = (typeof(_priv))(_base_priv); \
             if (_cond) { \
                 ucs_queue_del_iter(_queue, _iter); \
-                (void)_cb(ucs_container_of(_base_priv, uct_pending_req_t, priv), _arg); \
+                if ((_cb) != NULL) { \
```

Comment:
are you going to use it somewhere?

Reply:
yes, to implement `UCT_FLUSH_FLAG_CANCEL` support for TCP in the same way as it is done in other TLs - e.g. `uct_dc_mlx5_ep_flush()`

Reply:
can we use ucs_empty_function_return_XX instead?

Reply:
@yosefe yes, we can. But other transports can pass NULL. It would be better to align them. Maybe check if this callback is NULL in this macro. And if it is NULL, set it to empty_callback to avoid checking for NULL in each iteration. WDYT?

Reply:
in general i don't think we need to support NULL in pending_purge. wouldn't it cause memory leak?

Reply:
@yosefe yes, I agree with you. Looks like it is common UCT issue and could lead to memory leak. I think we should re-implement flush cancel in all transports to use user's specified cb from UCT iface params and release requests, WDYT?

Reply:
yes, but let's start with TCP

Reply:
@dmitrygx can we close this PR?

Reply:
> @dmitrygx can we close this PR?

sure, closing it